src/kdf.c

```diff
@@ -298,7 +298,7 @@ static int p11prov_hkdf_set_ctx_params(void *ctx, const OSSL_PARAM params[])
     /* can be multiple paramaters, which wil be all concatenated */
     for (p = OSSL_PARAM_locate_const(params, OSSL_KDF_PARAM_INFO); p != NULL;
          p = OSSL_PARAM_locate_const(p + 1, OSSL_KDF_PARAM_INFO)) {
-        void *ptr;
+        uint8_t *ptr;
```

Comment:
Why this change ?

Reply:
Might be better to cast later if it is a type issue.

Reply:
Thinking again, it is probably ok, nvm

Reply:
Pointer arithmetics on `void *` is GNU extension and not defined by C standard.

Reply:
Yeah, I realized that when I expanded the code below, I already forgotten this code :-D